### PR TITLE
Add content for the new building module - Closes #754

### LINKED ISSUE #754: Add building module to the manual
Parent issue: #751

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #754

### CODE REVIEW

File: modules/building/pages/contributing.adoc [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+:idprefix:
+// External URLs:
+:url_docs_staging: https://liskhq.github.io/lisk-docs
+:url_github_issues: https://github.com/LiskHQ/lisk-docs/issues
+// Project URLs:
+:url_staging: staging.adoc
+
+== Workflow
+
+. If needed, <<creating issues,create an issue>>, before working on it
```
Comment (chris529) [inline modules/building/pages/contributing.adoc]:
```suggestion
. If necessary, <<creating issues, create an issue>>, before working on it.
```

File: modules/building/pages/contributing.adoc [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+=== Creating issues
+
+Please create new issues concerning the Lisk documentation at {url_github_issues}.
+
+IMPORTANT: Avoid posting redundant issues: Check before, if someone has already opened an issue concerning the topic.
+
+* Templates:
+** Use the template `Bug` for any errors that you found inside the documentation.
+** Use template `Feature` for further ideas how to improve the documentation (e.g. structure changes of content or adding additional content).
+** Use template `Update` content that is outdated.
```
Comment (chris529) [inline modules/building/pages/contributing.adoc]:
```suggestion
** Use template `Update` for content that is outdated.
```

File: modules/building/pages/contributing.adoc [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+* Title: Choose a self-explaining, clear title
```
Comment (chris529) [inline modules/building/pages/contributing.adoc]:
```suggestion
* Title: Choose a clear self-explanatory title.
```

File: modules/building/pages/contributing.adoc [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+NOTE: If an issue affects multiple branches, it is always an <<epic, epic issue>>.
+
+//TODO: Update image
+//image:issue-templates.png[Lisk docs issue templates]
+
+[[epic]]
+=== Epics and child issues
+
+An epic issue is an issue that needs several PRs to be solved.
```
Comment (chris529) [inline modules/building/pages/contributing.adoc]:
```suggestion
An epic issue is an issue that needs several PRs for it to be solved.
```

File: modules/building/pages/contributing.adoc [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+git push origin 123-issue-title
+----
+
+== Projects
+
+== Pull requests (PRs)
+
+=== Creating a PR
+
+. On GitHub, open a PR with `docs-sdk` as base branch.
```
Comment (chris529) [inline modules/building/pages/contributing.adoc]:
```suggestion
. On GitHub, open a PR with `docs-sdk` as the base branch.
```

File: modules/building/pages/contributing.adoc [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+Check, if the issue is also linked on the right side panel.
+. Add the PR to the same project as the corresponding issue.
+. Click on btn:[Create pull request] to create the PR which is ready for review, or btn:[Create draft pull request] to create a PR that is still work in progress.
+. Request a reviewer, once the PR is ready from your side.
+
+== Preview on staging
+
+To verify the correct presentation of the changed content, the docs can be previewed on {url_docs_staging} page.
+
+Check the guide xref:{url_staging}[] for mor information how to deploy your branch on staging.
```
Comment (chris529) [inline modules/building/pages/contributing.adoc]:
```suggestion
Check the guide xref:{url_staging}[] for more information on how to deploy your branch on staging.
```

File: modules/building/pages/contributing.adoc [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+[CAUTION]
+====
+Please use only the side nav and the links inside the content to navigate in the stagin documentation.
```
Comment (chris529) [inline modules/building/pages/contributing.adoc]:
```suggestion
Please use only the side nav and the links inside the content to navigate in the staging documentation.
```

File: modules/building/pages/contributing.adoc [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+The next- and previous links at the bottom of the page generally don't work on staging.
```
Comment (chris529) [inline modules/building/pages/contributing.adoc]:
```suggestion
The next and previous links at the bottom of the page generally don't work on staging.
```

File: modules/building/pages/index.adoc [hunk 9]
```diff
@@ -1,3 +1,106 @@
 = Building the Lisk documenation
```
Comment (chris529) [inline modules/building/pages/index.adoc]:
```suggestion
= Building the Lisk documentation
```

File: modules/building/pages/index.adoc [hunk 10]
```diff
@@ -1,3 +1,106 @@
...
+:url_docs_staging: https://liskhq.github.io/lisk-docs
+:url_github_docs: https://github.com/LiskHQ/lisk-docs
+:url_gitlab_antora_extensions: https://gitlab.com/antora/antora-asciidoctor-extensions
+// Project URLs:
+:url_contributing: contributing.adoc
+:url_search: search.adoc
+:url_staging: staging.adoc
+:url_structure: structure.adoc
+
+Lisk uses {url_antora}[Antora^] as documentation generator.
```
Comment (chris529) [inline modules/building/pages/index.adoc]:
```suggestion
Lisk uses {url_antora}[Antora^] as a documentation generator.
```

File: modules/building/pages/index.adoc [hunk 11]
```diff
@@ -1,3 +1,106 @@
...
+where
+
+** `COMPONENT` can be either `sdk`, `core`, `protocol` or `service`.
+** `1.2.3` represents the respective version number of the corresponding product.
+
+=== Structure
+
+== UI branches
+
+The presentation of the documentation is stored completely separate form the actual content.
```
Comment (chris529) [inline modules/building/pages/index.adoc]:
```suggestion
The presentation of the documentation is stored completely separate from the actual content.
```

File: modules/building/pages/index.adoc [hunk 12]
```diff
@@ -1,3 +1,106 @@
...
+
+* `dev-antora`: This branch stores the Lisk docs UI in development.
```
Comment (chris529) [inline modules/building/pages/index.adoc]:
```suggestion
* `dev-antora`: This branch stores the Lisk docs UI used in development.
```

File: modules/building/pages/index.adoc [hunk 13]
```diff
@@ -1,3 +1,106 @@
...
+
+* ...the `site.yml` file, which is used to generate the Lisk Documentation using Antora.
+* ...the `searchdocs-scraper` folder, for creating the search index.
+* ...the `lib` folder, which stores the <<Asciidoc extensions>>, that are used in the Lisk documentation.
+
+NOTE: More information under xref:{url_structure}[], xref:{url_staging}[] and xref:{url_search}[].
+
+=== Asciidoc extensions
+
+The following Asciidoc extension are used currently:
```
Comment (chris529) [inline modules/building/pages/index.adoc:84]:
```suggestion
The following Asciidoc extension are used currently: 
```
I think something is missing here..?

File: modules/building/pages/releases.adoc [hunk 14]
```diff
@@ -1,39 +1,276 @@
...
 
-//(easiest to make an example here, for example 3.x to 4.x release of the sdk docs)
+:url_github_sdk_examples: https://github.com/LiskHQ/lisk-sdk-examples/tree/development
+:url_staging: https://liskhq.github.io/lisk-docs
 
+:url_contributing: contributing.adoc
+:url_search: search.adoc
+:url_styleguide: styleguide:index.adoc
 
+The steps that are necessary and/or recommended for updating the documentation to a new version are explained on this page.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
The steps that are both necessary and recommended for updating the documentation to a new version are explained on this page.
```

File: modules/building/pages/releases.adoc [hunk 15]
```diff
@@ -1,39 +1,276 @@
...
+
+== Explaining by example: Releasing the SDK 4.0.0 docs
+
+This guide uses the release of the SDK 4.0.0 docs as an example to illustrate the general steps that need to be done for a new release on a practical level.
+Lisk SDK version 3.0.2 is the previous latest version in this example.
+
+NOTE: The steps explained here are not specific for the Lisk SDK docs and do also apply for other documentation components like `core`, `protocol` and `service`.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
NOTE: The steps explained here are not specific for the Lisk SDK docs and also apply for other documentation components such as `core`, `protocol` and `service`.
```

File: modules/building/pages/releases.adoc [hunk 16]
```diff
@@ -1,39 +1,276 @@
...
+:sectnumlevels: 2
+[[preparing]]
 == Preparing the new docs
 
-* Learn about the upcoming changes by reading the changelog, looking at the github repo, the projects and the code.
-* Create a new project on lisk-docs for the new version, title: component + version, description: follow the existing examples
-* Create issues for all upcoming content changes in the docs and add them to the project.
+=== Learning
+
+Examples how to get knowledge about the new release:
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
Examples of how to acquire further knowledge about the new release:
```

File: modules/building/pages/releases.adoc [hunk 17]
```diff
@@ -1,39 +1,276 @@
...
+* Get onboarded internally (if provided)
+** via internal changelog
+** via STMs
+* Get onboarded on GitHub in the project description: https://github.com/LiskHQ/lisk-sdk/projects
+** Read all LIPs carefully, which are linked in the project description
++
+image:project-changelog.png[]
+* Check the corresponding GitHub project: https://github.com/LiskHQ/lisk-sdk/projects/76
+** Look at the epic issues
+** Go down to the child issues and PRs of certain epics of needed
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
** Navigate into the child issues and PRs of certain epics if necessary
```

File: modules/building/pages/releases.adoc [hunk 18]
```diff
@@ -1,39 +1,276 @@
...
+* description: follow the existing examples
+. xref:{url_contributing}[Create issues] for all upcoming content changes in the docs and add them to the project.
+New issues can still be added later, if necessary.
+
+==== Major release
+
+If a major version is released, a new content branch needs to be created for it.
+
+Create a new branch `docs-sdk-3.0.2`, on the base of `docs-sdk`.
 Like the version number indicates, this branch will store the docs that become outdated with the new release.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
As the version number indicates, this branch will store the docs that become outdated with the new release.
```

File: modules/building/pages/releases.adoc [hunk 19]
```diff
@@ -1,39 +1,276 @@
...
+.On branch `docs-sdk`
+[source,bash]
+----
+git checkout docs-sdk
+git pull origin docs-sdk
+git checkout -b docs-sdk-3.0.2 <1>
+git push origin docs-sdk-3.0.2 <2>
+git checkout docs-sdk <3>
+----
+<1> Switch to a new base branch for the SDK v3 docs
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
<1> Switch to a new base branch for the SDK v3 docs.
```

File: modules/building/pages/releases.adoc [hunk 20]
```diff
@@ -1,39 +1,276 @@
...
+<2> Push the new content branch to the lisk-docs repository
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
<2> Push the new content branch to the lisk-docs repository.
```

File: modules/building/pages/releases.adoc [hunk 21]
```diff
@@ -1,39 +1,276 @@
...
+<3> Go back to the `docs-sdk` branch
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
<3> Go back to the `docs-sdk` branch.
```

File: modules/building/pages/releases.adoc [hunk 22]
```diff
@@ -1,39 +1,276 @@
...
+. Example applications
++
+NOTE: These updates need to be done in the {url_github_sdk_examples}[lisk-sdk-examples^] repository.
+. Tutorials
+. Guides
+. Changelog
+
+== UI updates
+
+Use `dev_antora` as base branch for all pull requests.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
Use `dev_antora` as the base branch for all pull requests.
```

File: modules/building/pages/releases.adoc [hunk 23]
```diff
@@ -1,39 +1,276 @@
...
+`dev_antora` is used in the {url_staging}[lisk-docs staging environment^].
+
+* Update top navigation in `src/partials/header-content.hbs`:
+** Check, if any links are outdated and update or remove them.
+** Check, if any new links should be added.
+** If a new documentation component is released: The new component needs to be added to show dynamically in the top left.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
** If a new documentation component is released: The new component needs to be added in the top left side and shown dynamically.
```

File: modules/building/pages/releases.adoc [hunk 24]
```diff
@@ -1,39 +1,276 @@
...
+[[update_previous]]
+==== Update `antora.yml`
+
+IMPORTANT: This needs to be updated at the same time as on the <<update_latest,latest version branch>>.
+
+* Create a new issue to update the version of the previous version branch in `antora.yml`.
+* Change `master` to `3.0.2`.
+* Change `3.0.2 (latest)` to `3.0.2`.
+
+==== Updates on other branches like `docs-core-3.0.0-beta.0`
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
==== Updates on other branches such as `docs-core-3.0.0-beta.0`
```

File: modules/building/pages/releases.adoc [hunk 25]
```diff
@@ -1,39 +1,276 @@
...
+
+[NOTE]
+====
+Do not forget to check if you need to update in other components the references to pages in `docs-sdk-3.0.2` from `master` to `3.0.2`.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
Do not forget to check if it is necessary to perform any updates in other components, which relate to references within the applicable pages in the `docs-sdk-3.0.2` from `master` to `3.0.2`.
```

File: modules/building/pages/releases.adoc [hunk 26]
```diff
@@ -1,39 +1,276 @@
...
+----
+npm i lisk-sdk # before
+----
+
+[source,bash]
+----
+npm i lisk-sdk@3.0.2 # after
+----
+
+Do this for all Lisk Elements packages and Lisk Commander as well.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
Perform this for all Lisk Elements packages and also Lisk Commander.
```

File: modules/building/pages/releases.adoc [hunk 27]
```diff
@@ -1,39 +1,276 @@
...
+source antora/.env
+antora live-site.yml --fetch
+cd live
+git status
+git add .
+git commit -m "Update docs"
+git push origin live
+----
+
+Now, it is needed to ask the Devops team to update the documentation on lisk.io.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
At this point, it is necessary to ask the Devops team to update the documentation on lisk.io.
```

File: modules/building/pages/releases.adoc [hunk 28]
```diff
@@ -1,39 +1,276 @@
...
+They will then deploy the new content and flush the cache.
+
+=== Updating the search index
+
+Make sure Docker is started and that you are on branch `build`.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
Ensure that Docker is started and that you are on branch `build`.
```

File: modules/building/pages/releases.adoc [hunk 29]
```diff
@@ -1,39 +1,276 @@
...
+
+[source,bash]
+----
+cd searchdocs-scraper
+----
+
+First, make sure you have Docker running on your machine and that you can list images.
```
Comment (chris529) [inline modules/building/pages/releases.adoc]:
```suggestion
Firstly, ensure you have Docker running on your machine and that you can list images.
```

File: modules/building/pages/search.adoc [hunk 30]
```diff
@@ -0,0 +1,98 @@
...
+:url-repo: https://gitlab.com/liskhq/lisk-docs
+// External URLs:
+:url-algolia: https://www.algolia.com/doc/guides/getting-started/what-is-algolia/
+:url-docsearch-scraper-repo: https://github.com/algolia/docsearch-scraper
+:url-docsearch-scraper-docs: https://community.algolia.com/docsearch/run-your-own.html
+:url-pipenv: https://pipenv.readthedocs.io/en/latest/
+:url-doc-susemanager: https://github.com/SUSE/doc-susemanager/wiki/Setup-Algolia-Search-with-Antora
+:url-antora-search: https://gitlab.com/antora/antora-ui-default/issues/44
+
+This guide contains information for how to manage the search index for the Lisk documentation site.
```
Comment (chris529) [inline modules/building/pages/search.adoc]:
```suggestion
This guide contains information on how to manage the search index for the Lisk documentation site.
```

File: modules/building/pages/search.adoc [hunk 31]
```diff
@@ -0,0 +1,98 @@
...
+It documents where the search index is located and how to update it.
+
+== Overview
+
+The search index for the documentation is hosted by {url-algolia}[Algolia].
+The name of the index is `lisk-docs-search`.
+The index is populated by the {url-docsearch-scraper-repo}[docsearch scraper] (aka crawler).
+
+The sections below document the prerequisites for running the docsearch scraper and how to run it to update the index.
```
Comment (chris529) [inline modules/building/pages/search.adoc]:
```suggestion
The sections below documents the prerequisites for running the docsearch scraper and how to run it to update the index.
```

File: modules/building/pages/search.adoc [hunk 32]
```diff
@@ -0,0 +1,98 @@
...
+
+[subs=attributes+]
+....
+$ git clone {url-repo} &&
+cd "`basename $_`" &&
+git checkout build &&
+cd searchdocs-scraper
+....
+
+Staying in the `searchdocs-scraper` folder, also clone the {url-docsearch-scraper-repo} repository using git:
```
Comment (chris529) [inline modules/building/pages/search.adoc]:
```suggestion
Remain in the `searchdocs-scraper` folder, and also clone the {url-docsearch-scraper-repo} repository using git:
```

File: modules/building/pages/search.adoc [hunk 33]
```diff
@@ -0,0 +1,98 @@
...
+
+[subs=attributes+]
+ $ git clone {url-docsearch-scraper-repo} .
+
+Next, create an `.env` file in the `searchdocs-scraper` directory.
+This file is used to define the application ID (`APPLICATION_ID`) and write API key (`API_KEY`) for the scraper.
```
Comment (chris529) [inline modules/building/pages/search.adoc]:
```suggestion
This file is used to define the application ID (`APPLICATION_ID`) and write the API key (`API_KEY`) for the scraper.
```

File: modules/building/pages/search.adoc [hunk 34]
```diff
@@ -0,0 +1,98 @@
...
+ $ pipenv install --three
+
+Finally, you'll need the docsearch configuration file.
+This configuration file `config.json` is located in the `searchdocs-scraper` directory.
+
+You're now ready to run the scraper.
+
+== Usage
+
+Keep in mind that building the index takes several minutes because the scraper has to visit every page in the site.
```
Comment (chris529) [inline modules/building/pages/search.adoc]:
```suggestion
Keep in mind that building the index takes several minutes to complete, as the scraper has to visit every page in the site.
```

File: modules/building/pages/search.adoc [hunk 35]
```diff
@@ -0,0 +1,98 @@
...
+
+WARNING: The `docsearch docker:run` command will not be available if the APPLICATION_ID and API_KEY are not defined in the `.env` file.
+
+First, make sure you have Docker running on your machine and that you can list images.
```
Comment (chris529) [inline modules/building/pages/search.adoc]:
```suggestion
Firstly, ensure you have Docker running on your machine and that you can list images.
```

File: modules/building/pages/staging.adoc [hunk 36]
```diff
@@ -0,0 +1,74 @@
...
+- Checkout the `build` branch
+- Setup the working environment like described in xref:{url_structure}[]
+
+=== Antora installation
+
+ npm install --global @antora/cli@2.3 @antora/site-generator-default@2.3
+
+=== Algolia credentials
+
+To include the aloglia search box in the UI when building the documentation:
```
Comment (chris529) [inline modules/building/pages/staging.adoc]:
```suggestion
How to include the aloglia search box in the UI when building the documentation:
```

File: modules/building/pages/staging.adoc [hunk 37]
```diff
@@ -0,0 +1,74 @@
...
+  bundle:
+   url: ./dev-antora/build/ui-bundle.zip <2>
+   snapshot: true
+output:
+  dir: ./master <3>
+~
+----
+
+<1> All branch names listed under the `branches` key will be used by Antora to build the documentation content.
+<2> Defines the UI bundle that will be used by Antora to build the documentation
```
Comment (chris529) [inline modules/building/pages/staging.adoc]:
```suggestion
<2> Defines the UI bundle that will be used by Antora to build the documentation.
```

File: modules/building/pages/structure.adoc [hunk 38]
```diff
@@ -0,0 +1,60 @@
+= Repository structure and workflow
+// External URLs
+:url_github_docs: git@github.com:LiskHQ/lisk-docs.git
+:url_git_worktree: https://git-scm.com/docs/git-worktree
+// Project URLs
+
+It is highly recommended to use git worktrees when working with the Lisk documentation.
+
+This allows to conveniently work on different content branches and to deploy the docs on the go, without having to switch branches all the time.
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
This provides the option to conveniently work on different content branches and to deploy the docs on the go, without having to switch branches all the time.
```

File: modules/building/pages/structure.adoc [hunk 39]
```diff
@@ -0,0 +1,60 @@
...
+
+== Setting up the working environment for the first time
+
+=== Working with git worktrees
+
+Git `worktree` allows to have checked out multiple branches of one repository.
+Each branch is stored in folder with a name analog to the branch name.
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
Each branch is stored in a folder with a name analog to the branch name.
```

File: modules/building/pages/structure.adoc [hunk 40]
```diff
@@ -0,0 +1,60 @@
...
+cd lisk-docs
+git checkout build
+git worktree add dev-antora dev-antora
+git worktree add antora antora
+git worktree add docs-sdk docs-sdk
+git worktree add docs-service docs-service
+git worktree add docs-core docs-core
+```
+
+After setting up the worktree, the build branch should have several new folder and should look similar to this:
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
After setting up the worktree, the build branch should have several new folders, and should look similar to the structure shown below:
```

File: modules/building/pages/structure.adoc [hunk 41]
```diff
@@ -0,0 +1,60 @@
...
+
+It is highly recommended to use git worktrees when working with the Lisk documentation.
+
+This allows to conveniently work on different content branches and to deploy the docs on the go, without having to switch branches all the time.
+
+== Setting up the working environment for the first time
+
+=== Working with git worktrees
+
+Git `worktree` allows to have checked out multiple branches of one repository.
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
Git `worktree` offers the possibility to check out multiple branches of one repository.
```

File: modules/building/pages/structure.adoc [hunk 42]
```diff
@@ -0,0 +1,60 @@
...
+├── docs-service <5>
+├── lib <6>
+├── searchdocs-scraper <7>
+├── master <8>
+├── live <9>
+├── live-site.yml <10>
+└── site.yml <11>
+----
+
+<1> Stores the `antora` branch (UI)
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<1> Stores the `antora` branch (UI).
```

File: modules/building/pages/structure.adoc [hunk 43]
```diff
@@ -0,0 +1,60 @@
...
+<2> Stores the `dev-antora` branch (UI)
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<2> Stores the `dev-antora` branch (UI).
```

File: modules/building/pages/structure.adoc [hunk 44]
```diff
@@ -0,0 +1,60 @@
...
+<3> Stores the `docs-core` branch (Content)
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<3> Stores the `docs-core` branch (Content).
```

File: modules/building/pages/structure.adoc [hunk 45]
```diff
@@ -0,0 +1,60 @@
...
+<4> Stores the `docs-sdk` branch (Content)
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<4> Stores the `docs-sdk` branch (Content).
```

File: modules/building/pages/structure.adoc [hunk 46]
```diff
@@ -0,0 +1,60 @@
...
+<5> Stores the `docs-service` branch (Content)
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<5> Stores the `docs-service` branch (Content).
```

File: modules/building/pages/structure.adoc [hunk 47]
```diff
@@ -0,0 +1,60 @@
...
+<6> Stores the extentions for Antora, which are used in the playbook files.
+<7> Contains all files relates to the algolia docs scraper
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<7> Contains all files relates to the algolia docs scraper.
```

File: modules/building/pages/structure.adoc [hunk 48]
```diff
@@ -0,0 +1,60 @@
...
+<8> Stores the `master` branch, which hold the staging documentation (HTML)
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<8> Stores the `master` branch, which holds the staging documentation (HTML).
```

File: modules/building/pages/structure.adoc [hunk 49]
```diff
@@ -0,0 +1,60 @@
...
+<9> Stores the `live` branch, which hold the production documentation (HTML)
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<9> Stores the `live` branch, which holds the production documentation (HTML).
```

File: modules/building/pages/structure.adoc [hunk 50]
```diff
@@ -0,0 +1,60 @@
...
+<10> The playbook file for the production documentation
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<10> The playbook file for the production documentation.
```

File: modules/building/pages/structure.adoc [hunk 51]
```diff
@@ -0,0 +1,60 @@
...
+<11> The playbook file for the staging documentation
```
Comment (chris529) [inline modules/building/pages/structure.adoc]:
```suggestion
<11> The playbook file for the staging documentation.
```

File: modules/building/pages/ui.adoc [hunk 52]
```diff
@@ -0,0 +1,179 @@
...
+  bundle:
+    url: https://github.com/LiskHQ/lisk-docs/raw/antora/build/ui-bundle.zip
+#   url: ./dev-antora/build/ui-bundle.zip
+    snapshot: true
+----
+
+== Development Quickstart
+
+This section offers a basic tutorial to teach you how to set up the default UI project, preview it locally, and bundle it for use with Antora.
+A more comprehensive can be found in the documentation at {url-antora-docs}.
```
Comment (chris529) [inline modules/building/pages/ui.adoc]:
```suggestion
A more comprehensive description can be found in the documentation at {url-antora-docs}.
```

File: modules/building/pages/ui.adoc [hunk 53]
```diff
@@ -0,0 +1,179 @@
...
+
+=== Prerequisites
+
+To preview and bundle the default UI, you need the following software on your computer:
```
Comment (chris529) [inline modules/building/pages/ui.adoc]:
```suggestion
To preview and bundle the default UI, you will need to have the following software installed on your computer:
```

File: modules/building/pages/ui.adoc [hunk 54]
```diff
@@ -0,0 +1,179 @@
...
+
+* {url-git}[git] (command: `git`)
+* {url-nodejs}[Node.js] (commands: `node` and `npm`)
+* {url-gulp}[Gulp CLI] (command: `gulp`)
+
+==== git
+
+First, make sure you have git installed.
```
Comment (chris529) [inline modules/building/pages/ui.adoc]:
```suggestion
Firstly, ensure you have git installed.
```

File: modules/building/pages/ui.adoc [hunk 55]
```diff
@@ -0,0 +1,179 @@
...
+
+ $ git --version
+
+If not, {url-git-dl}[download and install] the git package for your system.
+
+==== Node.js
+
+Next, make sure that you have Node.js installed (which also provides npm).
```
Comment (chris529) [inline modules/building/pages/ui.adoc]:
```suggestion
Next, ensure that you have Node.js installed (which also provides npm).
```

File: modules/building/pages/ui.adoc [hunk 56]
```diff
@@ -0,0 +1,179 @@
...
+
+Once you've installed nvm, open a new terminal and install Node.js 10 using the following command:
+
+ $ nvm install 10
+
+You can switch to this version of Node.js at any time using the following command:
+
+ $ nvm use 10
+
+To make Node.js 10 the default in new terminals, type:
```
Comment (chris529) [inline modules/building/pages/ui.adoc]:
```suggestion
To make Node.js 10 the default in all new terminals, type the following:
```

File: modules/building/pages/ui.adoc [hunk 57]
```diff
@@ -0,0 +1,179 @@
...
+
+ $ nvm alias default 10
+
+Now that you have Node.js installed, you can proceed with installing the Gulp CLI.
+
+==== Gulp CLI
+
+You'll need the Gulp command-line interface (CLI) to run the build.
+The Gulp CLI package provides the `gulp` command which, in turn, executes the version of Gulp declared by the project.
```
Comment (chris529) [inline modules/building/pages/ui.adoc]:
```suggestion
The Gulp CLI package provides the `gulp` command, which in turn executes the version of Gulp declared by the project.
```

File: modules/building/pages/ui.adoc [hunk 58]
```diff
@@ -0,0 +1,179 @@
...
+
+You can install the Gulp CLI globally (which resolves to a location in your user directory if you're using nvm) using the following command:
+
+ $ npm install -g gulp-cli
+
+Verify the Gulp CLI is installed and on your PATH by running:
```
Comment (chris529) [inline modules/building/pages/ui.adoc]:
```suggestion
Verify the Gulp CLI is installed and in your PATH by running this command:
```

File: modules/building/pages/ui.adoc [hunk 59]
```diff
@@ -0,0 +1,179 @@
...
+
+=== Initialize the UI Project
+
+Use npm to install the project's dependencies inside the project.
+In your terminal, execute the following command:
+
+ $ npm install
+
+This command installs the dependencies listed in [.path]_package.json_ into the [.path]_node_modules/_ folder inside the project.
+This folder does not get included in the UI bundle and should _not_ be committed to the source control repository.
```
Comment (chris529) [inline modules/building/pages/ui.adoc:113]:
```suggestion
This folder does not get included in the UI bundle and should _not_ be committed to the source control repository.
```
Are these underscores after 'should' and 'not' in the above sentence correct?

File: modules/building/pages/ui.adoc [hunk 60]
```diff
@@ -0,0 +1,179 @@
...
+
+ $ yarn
+====
+
+=== Preview the UI
+
+The default UI project is configured to preview offline.
+The files in the [.path]_preview-src/_ folder provide the sample content that allow you to see the UI in action.
+In this folder, you'll primarily find pages written in AsciiDoc.
+These pages provide a representative sample and kitchen sink of content from the real site.
```
Comment (chris529) [inline modules/building/pages/ui.adoc:127]:
```suggestion
These pages provide a representative sample and a myriad of content from the real site.
```

File: modules/building/pages/ui.adoc [hunk 61]
```diff
@@ -0,0 +1,179 @@
...
+==== Source Maps
+
+The build consolidates all the CSS and client-side JavaScript into combined files, [.path]_site.css_ and [.path]_site.js_, respectively, in order to reduce the size of the bundle.
+{url-source-maps}[Source maps] correlate these combined files with their original sources.
+
+This "`source mapping`" is accomplished by generating additional map files that make this association.
+These map files sit adjacent to the combined files in the build folder.
+The mapping they provide allows the debugger to present the original source rather than the obfuscated file, an essential tool for debugging.
+
+In preview mode, source maps are enabled automatically, so there's nothing you have to do to make use of them.
```
Comment (chris529) [inline modules/building/pages/ui.adoc]:
```suggestion
In preview mode, source maps are enabled automatically, so no other requirements are necessary to use them.
```

File: modules/building/pages/ui.adoc [hunk 62]
```diff
@@ -0,0 +1,179 @@
...
+If you need to package the UI so you can use it to generate the documentation site locally, run the following command:
+
+ $ gulp bundle
+
+If any errors are reported by lint, you'll need to fix them.
+
+When the command completes successfully, the UI bundle will be available at [.path]_build/ui-bundle.zip_.
+You can point Antora at this bundle using the `--ui-bundle-url` command-line option.
+
+If you have the preview running, and you want to bundle without causing the preview to be clobbered, use:
```
Comment (chris529) [inline modules/building/pages/ui.adoc:159]:
```suggestion
If you have the preview running, and you want to bundle without causing the preview to be clobbered, use:
```
Am not sure 'clobbered' is the right word here..? Do you mean damaged or impaired..?

**Review by chris529 — COMMENTED** [review]
Looks very good Mona.. It is just all small 'stuff' 👍

File: modules/building/pages/releases.adoc (hunk 25 above)
Comment (Tschakki) [reply modules/building/pages/releases.adoc]:
```suggestion
Do not forget to check if it is necessary to perform any updates in other components, which relate to references within the applicable pages.
```

**Pull request merged**